integrations/Mouseflow/browser.js

```diff
@@ -0,0 +1,105 @@
+/* eslint-disable no-underscore-dangle */
+/* eslint-disable class-methods-use-this */
+import get from "get-value";
+import { NAME } from "./constants";
+import logger from "../../utils/logUtil";
+import ScriptLoader from "../ScriptLoader";
+import setCustomVariables from "./utils";
+import { isDefinedAndNotNull } from "../utils/commonUtils";
+
+class Mouseflow {
+  constructor(config) {
+    this.websiteId = config.websiteId;
+    this.name = NAME;
+  }
+
+  init() {
+    logger.debug("===In init mouseflow===");
+    window._mfq = window._mfq || [];
+    ScriptLoader(
+      "mouseflow-integraions",
+      `https://cdn.mouseflow.com/projects/${this.websiteId}.js`
+    );
+  }
+
+  isLoaded() {
+    logger.debug("===In isLoaded mouseflow===");
+    return !!window.mouseflow && typeof window.mouseflow === "object";
+  }
+
+  isReady() {
+    logger.debug("===In isReady mouseflow===");
+    return !!window._mfq;
+  }
+
+  /*
+   * Add tags
+   * Set custom Variables
+   * Ref: https://js-api-docs.mouseflow.com/#setting-a-custom-variable
+   */
+  addTags(message) {
+    const tags = get(message, `integrations.${NAME}.tags`);
+    if (isDefinedAndNotNull(tags)) {
+      setCustomVariables(tags);
+    }
+  }
+
+  /**
+   * Identify.
+   * for supporting userId or email
+   * Ref: https://js-api-docs.mouseflow.com/#identifying-a-user
+   * for supporting user traits and tags
+   * Ref: https://js-api-docs.mouseflow.com/#setting-a-custom-variable
+   * @param {Identify} identify
+   */
+  identify(rudderElement) {
+    const { message } = rudderElement;
+    const { traits } = message.context;
+    const email =
+      get(message, "context.traits.email") || get(message, "traits.email");
+    const userId = message.userId || email;
+    window._mfq.push(["stop"]);
+    if (userId) window._mfq.push(["identify", userId]);
+    window.mouseflow.start();
+    setCustomVariables(traits);
```

Comment:
Did we check if `traits` is undefined how the function `setcustomVariables` is handling it?

Reply:
updated